examples/notebooks/beam-ml/rag_usecase/redis_connector.py

```diff
@@ -0,0 +1,399 @@
+import logging
+
+logging.root.setLevel(logging.INFO)
+logger = logging.getLogger(__name__)
+
+import redis
+
+__all__ = ['InsertDocInRedis','WriteToRedis']
+
+
+class Document(typing.NamedTuple):
+    id: str
+    url: str
+    title: str
+    text: str
+    
+coders.registry.register_coder(Document, coders.RowCoder)
+
+
+"""This module implements IO classes to read write data on Redis.
+
+
+Insert Doc in Redis:
+-----------------
+:class:`InsertDocInRedis` is a ``PTransform`` that writes key and values to a
+configured sink, and the write is conducted through a redis pipeline.
+
+The ptransform works by getting the first and second elements from the input,
+this means that inputs like `[k,v]` or `(k,v)` are valid.
+
+Example usage::
+
+  pipeline | WriteToRedis(host='localhost',
+                          port=6379,
+                          batch_size=100)
+
+
+No backward compatibility guarantees. Everything in this module is experimental.
+"""
+@deprecated(since='v.1')
+class InsertDocInRedis(PTransform):
+    """InsertDocInRedis is a ``PTransform`` that writes a ``PCollection`` of
+    key, value tuple or 2-element array into a redis server.
+    """
+
+    def __init__(self, host=None, port=None, command=None, batch_size=100):
+        """
+
+        Args:
+        host (str, ValueProvider): The redis host
+        port (int, ValueProvider): The redis port
+        batch_size(int, ValueProvider): Number of key, values pairs to write at once
+        command : command to be executed with redis client
+
+        Returns:
+        :class:`~apache_beam.transforms.ptransform.PTransform`
+
+        """
+        if not isinstance(host, (str, unicode, ValueProvider)):
+            raise TypeError(
+                '%s: host must be string, or ValueProvider; got %r instead'
+            ) % (self.__class__.__name__, (type(host)))
+
+        if not isinstance(port, (int, ValueProvider)):
+            raise TypeError(
+                '%s: port must be int, or ValueProvider; got %r instead'
+            ) % (self.__class__.__name__, (type(port)))
+
+        if not isinstance(port, (int, ValueProvider)):
+            raise TypeError(
+                '%s: batch_size must be int, or ValueProvider; got %r instead'
+            ) % (self.__class__.__name__, (type(batch_size)))
+
+        if isinstance(host, (str, unicode)):
+            host = StaticValueProvider(str, host)
+
+        if isinstance(port, int):
+            port = StaticValueProvider(int, port)
+
+        if isinstance(command, int):
+            command = StaticValueProvider(str, command)
+
+        if isinstance(batch_size, int):
+            batch_size = StaticValueProvider(int, batch_size)
+
+        self._host = host
+        self._port = port
+        self._command = command
+        self._batch_size = batch_size
+
+    def expand(self, pcoll):
+        return pcoll \
+               | Reshuffle() \
+               | beam.ParDo(_InsertDocInRedisFn(self._host,
```

Comment:
These should have names like:

```
| "Reshuffle for Redis Insert" >> Reshuffle()
| "Insert document into Redis" >> beam.ParDo(_InsertDocInRedisFn(self._host, ....
```

Reply:
I added the labels in the pipeline as you suggested but still it's giving the same issue.